### PR TITLE
Prevent adding reusable blocks to the basket if the parent post has the same source lang as the target lang

### PR DESCRIPTION
**Pull request by strategio**
We will filter the target langs before to scan the post looking for
reusable blocks.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6689

### CODE REVIEW

**Comment by strategio** [comment]
@OnTheGoSystems/wpml please review.

**Review by kagg-design — DISMISSED** [review]
+1 @strategio

File: src/reusable-blocks/class-manage-basket.php [hunk 1]
```diff
@@ -62,18 +62,40 @@ private function extractAddedPostElements( array $data ) {
...
-					&& 'post' === $item['type']
+					! isset( $item['checked'], $item['type'] )
+					|| 'post' !== $item['type']
 				) {
-					return new BasketElement(
-						(int) $item['checked'],
-						$source_lang,
-						$target_langs
-					);
+					return null;
```
Comment (andreasciamanna) [inline src/reusable-blocks/class-manage-basket.php:68]: Another approach could be something like that:

```php
return \wpml_collect( $data['post'] )
	->filter( function ( $item ) {
		return isset( $item['checked'], $item['type'] ) && 'post' !== $item['type'];
	} )
	->map(
		function ( $item ) use ( $source_lang, $target_langs ) {
			$target_langs = $this->filterTargetLangs( $target_langs, $item['checked'] );
			if ( ! $target_langs ) {
				return null;
			}
			return new BasketElement(
				(int) $item['checked'],
				$source_lang,
				$target_langs
			);
		}
	)
	->filter();
```

I'm a bit confused by the purpose of `filterTargetLangs`.

I think that with some changes in the code, you could move this logic on the same filter callback (if not even an additional one) before `map` and remove the last `->filer()`.

**Comment by andreasciamanna** [comment]
I think the logic is a bit overcomplicated, but it seems to do the work.

**Review by andreasciamanna — APPROVED** [review]

File: src/reusable-blocks/class-manage-basket.php (hunk 1 above)
Comment (strategio) [reply src/reusable-blocks/class-manage-basket.php:68]: @andreasciamanna, we need to compare the `$target_langs` against each `$item['checked']` (the post ID) to remove a target lang which would be the same as the post source lang.

Even if I move some code to the first `filter` (as you did), we still need to apply `filterTargetLangs` in the `map`, and so the last `filter` is still needed.

**Review by andreasciamanna — DISMISSED** [review]

**Pull request merged**